Add views to support Payments by Provider and Payment Provider Report…

kevinpostlewaite opened the pull request:
So many views 'cause MySQL does not permit subqueries within a view.

pierre commented:
:+1: 

One suggestion: could you add a README in your subdirectory (maybe something like https://github.com/killbill/killbill-analytics-plugin/blob/master/src/main/resources/reports/README.md), to explain a little bit the purpose of the report and/or how to run it?

kevinpostlewaite commented:
Didn't realize that this pull request was still open. Added READMEs and also modified the sanity.sql to:
1. correct a few queries that assumed that all payments are invoice_payments
2. add some new sanity queries written recently

kevinpostlewaite commented:
I also have some sanity queries for CyberSource and payU specific tables that I have not yet committed. Previously (for Zuora, PayPal) we put these into sanity.sql and commented them out. Does that sound good to you?

pierre commented:
> Added READMEs

Thanks!

> Previously (for Zuora, PayPal) we put these into sanity.sql and commented them out. Does that sound good to you?

I think this won't scale anymore. What do you think of creating plugin specific sub-directories under src/main/resources/sanity instead (e.g. src/main/resources/sanity/payu, etc.)?

kevinpostlewaite commented:
Good idea: I basically took your suggestion. I added a plugin_sanity directory under sanity and put a plugin-specific directory for each plugin there. I pulled the Zuora specific sanity out of the main sanity query and added my payU and CyberSource sanity queries.

pierre commented on src/main/resources/sanity/sanity.sql:
```diff
@@ -1237,3 +1187,861 @@ join (
+group by 1,2
+;
+
+select 'L2: Validate consistency between base payment_transactions tables and anlytics_payment_* tables' as sanity_query_name;select '' as sanity_query_name;
```
Typo? (second, empty, select)

pierre commented:
Looking good! I'll merge it once the typo is fixed.

kevinpostlewaite commented:
Fixed: thanks for catching!
